clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala

```diff
@@ -22,6 +22,41 @@ object StorageClientType extends Enumeration {
 
 private object ApiClient {
 
+  val clients = collection.mutable.Map[String, api.ApiClient]()
+
+  /** @return an ApiClient, reusing an existing one for this URL if possible.
+   */
+  def getApiClient(
+      apiUrl: String,
+      accessKey: String,
+      secretKey: String,
+      connectionTimeoutSec: String = "",
+      readTimeoutSec: String = ""
+  ): api.ApiClient = this.synchronized {
+    clients.get(apiUrl) match {
+      case Some(client) => client
+      case None => {
+        val FROM_SEC_TO_MILLISEC = 1000
+
+        val client = new api.ApiClient
+        client.setUsername(accessKey)
+        client.setPassword(secretKey)
+        client.setBasePath(apiUrl.stripSuffix("/"))
+        if (connectionTimeoutSec != null && !connectionTimeoutSec.isEmpty) {
+          val connectionTimeoutMillisec = connectionTimeoutSec.toInt * FROM_SEC_TO_MILLISEC
```

Comment:
(Not introduced in this PR, and to fix later:)  This might have a nasty failure mode.  When the configuration option is malformatted, the `String.toInt` call here is (probably) the first time that we discover it.  So this configuration error is reported as a stack trace from somewhere inside the driver, rather than during startup.  It will look a bit like this:
```
  at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
  at java.base/java.lang.Integer.parseInt(Integer.java:652)
  at java.base/java.lang.Integer.parseInt(Integer.java:770)
[...]
```
The error message will not say which configuration parameter was wrong -- read or connect timeouts; the user will have to open this file to determine that.

So flags should be parsed during startup.  I think we really want to fix this because it could be painful for users.

Reply:
Thanks! I added this to https://github.com/treeverse/lakeFS/issues/3989